weave/integrations/bedrock/bedrock_sdk.py

```diff
@@ -0,0 +1,151 @@
+from typing import TYPE_CHECKING, Any, Optional, Callable
+
+import weave
+from weave.trace.weave_client import Call
+from weave.trace.op_extensions.accumulator import add_accumulator, _IteratorWrapper
+from weave.trace.patcher import MultiPatcher, SymbolPatcher
+
+if TYPE_CHECKING:
+    from botocore.client import BaseClient
+
+
+def bedrock_on_finish_converse(
+    call: Call, output: Any, exception: Optional[BaseException]
+) -> None:
+    model_name = call.inputs["modelId"]
+    usage = {model_name: {"requests": 1}}
+    summary_update = {"usage": usage}
+    if output:
+        tokens_metrics = {
+            "prompt_tokens": output["usage"]["inputTokens"],
+            "completion_tokens": output["usage"]["outputTokens"],
+            "total_tokens": output["usage"]["totalTokens"],
+        }
+        usage[model_name].update(tokens_metrics)
+    if call.summary is not None:
+        call.summary.update(summary_update)
+
+
+def bedrock_on_finish_invoke(
+    call: Call, output: Any, exception: Optional[BaseException]
+) -> None:
+    model_name = call.inputs["modelId"]
+    usage = {model_name: {"requests": 1}}
+    summary_update = {"usage": usage}
+
+    if output and "ResponseMetadata" in output:
+        headers = output["ResponseMetadata"]["HTTPHeaders"]
+        prompt_tokens = int(headers.get("x-amzn-bedrock-input-token-count", 0))
+        completion_tokens = int(headers.get("x-amzn-bedrock-output-token-count", 0))
+        total_tokens = prompt_tokens + completion_tokens
+
+        tokens_metrics = {
+            "prompt_tokens": prompt_tokens,
+            "completion_tokens": completion_tokens,
+            "total_tokens": total_tokens,
+        }
+        usage[model_name].update(tokens_metrics)
+
+
+def postprocess_inputs(inputs: dict[str, Any]) -> dict[str, Any]:
```

Comment:
I have an upcoming PR that will let users pass `weave.op` kwargs through to the integrations.  Let's sync on this

Reply:
ok